docs/proposals/powderhouse-errors.md

```diff
@@ -0,0 +1,68 @@
+# Proposal for error handling in Powderhouse
+
+The same `Error` type will be used for parse errors and subsystem errors, including validation. It will also be available for use by invocations.
+
+`ParseResult` will contain collection of `CliError` objects that are created during parsing. This will include both argument specific errors like type conversion and arity, and general errors like response file issues. Errors for a specific argument or option will be available from its `ValueResult`.
+
+## API
+
+* `CliDiagnosticDescriptor`
+  *`string DiagnosticId`
+  * `string DefaultMessage` (to be used in `String.Format()`)
+  * `Uri HelpUri`
+  * `Severity DefaultSeverity`
+
+* `CliError`
+  * `public string DiagnosticId`
+  * `public Location Location` (which includes Symbol and Text to allow messages to be what user typed)
+  * `public string? Message` (defaults to Diagnostic)
```

Comment:
Unsure but I think you might mean `(defaults to DefaultMessage)` rather than `(defaults to Diagnostic)`